src/XMakeTasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -592,4 +599,170 @@ private void FreeAsmMeta(IntPtr asmMetaPtr)
             }
         }
     }
+
+    /// <summary>
+    /// Managed implementation of a reader for getting the runtime version of an assembly
+    /// </summary>
+    static class ManagedRuntimeVersionReader
+    {
+        class HeaderInfo
+        {
+            public uint VirtualAddress;
+            public uint Size;
+            public uint FileOffset;
+        }
+
+        /// <summary>
+        /// Given a path get the CLR runtime version of the file
+        /// </summary>
+        /// <param name="path">path to the file</param>
+        /// <returns>The CLR runtime version or empty if the path does not exist or the file is not an assembly.</returns>
+        public static string GetRuntimeVersion(string path)
```

Comment:
Would you mind including a pointer to the PE header spec (and version) you used? That would be helpful for future maintenance.